docs/design_doc.md

```diff
@@ -0,0 +1,443 @@
+
+### Elements
+
+We model a spatial dataset as a composition of distinct element types. The elements correspond to:
+
+-   Pixel-based images
+-   Regions of interest
+    -   Shapes (such as polygons, circles, ...)
+    -   Pixel masks (such as segmentation masks), aka Labels
+-   Points (such as transcript locations, point clouds, ...)
+-   Tables of annotations (initally, these are annotations on the regions of interest)
+
+Each of these elements should be useful by itself, and in combination with other relevant elements. All elements are stored in the Zarr container in hierarchy store that MAY be flat. `SpatialData` will nonetheless be able to read arbitrary hierarchies and make sense of them based on Zarr groups metadata and coordinate systems.
+
+By decomposing the data model into building blocks (i.e. Elements) we support operations that do not otherwise fit this model. For instance, we may have data where regions have not been defined yet, or are just dealing with images and points.
+
+#### Assumptions
+
+_SpatialData_ closely follows the OME-NGFF specifications and therefore much of its assumptions are inherited from it. Extra assumptions will be discussed with the OME-NGFF community and adapted to the community-agreed design. The key assumptions are the following:
+
+-   `Images`, `Labels`, `Points`, and `Shapes` MUST have one or more _coordinate systems_ and _coordinate transforms_.
+-   `Tables` CAN NOT have a _coordinate system_ or _coordinate transforms_.
+-   `Labels` and `Shapes` are both instances of `Regions`.
+-   `Regions` are `Elements` and they MAY be annotated by `Tables`.
+-   `Points` MAY be annotated with `Tables`.
+-   `Tables` CAN NOT be annotated by other `Tables`.
+
+#### Images
+
+Images of a sample. Should conform to the [OME-NGFF concept of an image](https://ngff.openmicroscopy.org/latest/#image-layout).
+
+Images are ndimensional arrays where each element of an array is a pixel of an image. These arrays have labelled dimensions which correspond to:
+
+-   Spatial dimensions (height and width).
+-   Imaging or feature channels.
+-   Z-stacks.
+
+We will support also time-point axes in the future. Furthermore, due to NGFF specs v0.5, such axes will not have name constraints (although they do for first iteration due to NGFF specs v0.4).
+
+The image object itself builds on prior art in image analysis, in particular the [xarray library](https://docs.xarray.dev/en/stable/).
+
+Images have labelled dimensions, and coordinate transformations. These transformations are used to map between pixel space and physical space, and between different physical spaces.
+
+For computational efficiency, images can have a pyramidal or multiscale format. This is implemented as an [xarray datatree](https://github.com/xarray-contrib/datatree).
```

Comment:
think we could add a reference to spatial-image but maybe only later on?

Reply:
I added a reference here. We can revise once we have more clarity on if/how we will depend on `spatial-image`